### DISCUSSION TITLE
xpra shadow and sreen lock

### DISCUSSION THREAD

**Discussion by Cigaes** [Xpra-org/xpra#4080] [Q&A]
Hi.

I had the following plan:
1. lock my workstation and go to a boring meeting;
2. in the boring meeting on my laptop: ssh workstation xpra shadow & xpra attach ssh:workstation

It works, except what xpra shows me is the locked screen of the workstation. Of course. You would have guessed.

I wonder if there is a way to make it work. For example an option to let xpra ignore the screen locker itself and only forward the other windows. Or a screen locker that supports xpra shadow to make it possible (I am currently using either i3lock or xtrlock bot I can consider changing).

Note 1: Starting a normal Xpra server on the workstation and attaching it locally, and later re-attaching it from the boring meeting when only the local screen of the workstation is locked, that would work. But it is not an option for me since Xpra is not transparent enough for a day-to-day use, only for exceptional circumstances that have already sub-standard comfort.

Note 2: Not locking my workstation or unlocking it remotely is not an option either.

Thanks in advance.

**Comment by totaam** [comment]
> in the boring meeting on my laptop: `ssh workstation xpra shadow & xpra attach ssh:workstation`

What's wrong with `xpra shadow ssh://workstation/` ?

>  Or a screen locker that supports xpra shadow

What would you want it to do?

>  or unlocking it remotely is not an option either.

Then I am afraid that it cannot be done.
`xpra shadow` works using screenscraping, it has no way of grabbing things that aren't shown, sorry.

**Comment by Cigaes** [comment]
> What's wrong with `xpra shadow ssh://workstation/` ?

Dunno, I had not checked if the shortcut was implemented.

> > Or a screen locker that supports xpra shadow
> What would you want it to do?

Cooperate with xpra somehow to let shadow work on a locked screen. I am not familiar with the internals, I cannot be more accurate.

> Then I am afraid that it cannot be done. `xpra shadow` works using screenscraping, it has no way of grabbing things that aren't shown, sorry.

I am fine if the screen is locked but not obscured, like xtrlock does. xtrlock only grabs the keyboard and mouse and changes the mouse pointer, the current contents of the screen is visible locally and properly appears on xpra shadow. So if xpra were able to bypass the grabbing, that would solve my use case.

I can consider patching xtrlock to have it grab only the physical devices through XInput, leaving the XTEST pointer and keyboard alone. Would this work with xpra?

**Comment by totaam** [comment]
> I can consider patching xtrlock to have it grab only the physical devices through XInput, leaving the XTEST pointer and keyboard alone. Would this work with xpra?

Yes, that would work.
